apis/externalsecrets/v1alpha1/pushsecret_types.go

```diff
@@ -105,7 +105,12 @@ type PushSecretSecret struct {
 	// +kubebuilder:validation:MinLength:=1
 	// +kubebuilder:validation:MaxLength:=253
 	// +kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
-	Name string `json:"name"`
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// SecretSelector chooses secrets using a labelSelector.
+	// +optional
+	SecretSelector *metav1.LabelSelector `json:"secretSelector,omitempty"`
```

Comment:
Maybe name this `selector` in favour of `secretSelector`. The secret bit is already implied :thinking: 

or alternatively we can consider embedding `matchLabels` and `matchExpressions` here as `ps.spec.selector.secret.matchLabels` and  `ps.spec.selector.secret.matchExpressions` respectively.


I'm leaning towards the latter. WDYT?

Reply:
Makes sense indeed!

Reply:
Having both embedded makes this check a bit convoluted:

```go
	case ps.Spec.Selector.Secret != nil && ps.Spec.Selector.Secret.Selector != nil:
		labelSelector, err := metav1.LabelSelectorAsSelector(ps.Spec.Selector.Secret.Selector)
		if err != nil {
			return nil, err
		}

		var secretList v1.SecretList
		err = r.List(ctx, &secretList, &client.ListOptions{LabelSelector: labelSelector})
		if err != nil {
			return nil, err
		}

		return secretList.Items, err
	}

```

Because now you need to see if one or the other is defined instead of checking if label selector is defined.

But I agree to the rename. :)